### ISSUE TITLE
Provide default krb5.conf or extend migration docs

### ISSUE THREAD

**Issue by hwanders** [keycloak/keycloak#32527]
### Description

When upgrading my Keycloak 20 to the most recent (I know, that was late), SSO with Kerberos stopped working.
TL;DR: The default `/etc/krb5.conf` is not provided since Keycloak 21 which was necessary for our configuration.

I think you should
* either provide a default `/etc/krb5.conf` (as it was before)
* or documented the necessity to add one in the migration guide (for Keycloak 21)

### Motivation

- We need to ensure compatibility with an old Kerberos server using the RC4-HMAC algorithm
- To achieve this we created a custom image based on yours, which adds a `crypto-policies` file to `/etc/krb5.conf.d/` file allowing weak algorithms
- This worked fine in Keycloak 20 but does not in Keycloak 25

### Details

I have analyzed the problem and found the reason:
- Since Keycloak 21 you seem to use Red Hat's UBI *micro* image as base image (see [this Dockerfile](https://github.com/keycloak/keycloak/blob/21.0.0/quarkus/container/Dockerfile))
- Previously you used the UBI *minimal* image (see [this earlier commit](https://github.com/keycloak/keycloak/commit/610e3044adea39003fdb256585a74d9adea2cd12#diff-95f281de08964d27b4569a023fe6b0561731e2f1bb701e05608d411edbc5315a) where you switched from `ubi8-minimal` to `ubi8-micro`)
- Unfortunately the micro image variant does not include the package `krb5-libs`
  - compare packages in [ubi9 micro](https://catalog.redhat.com/software/containers/ubi9/ubi-micro/615bdf943f6014fa45ae1b58?container-tabs=packages) and [ubi9 minimal](https://catalog.redhat.com/software/containers/ubi9-minimal/61832888c0d15aff4912fe0d?container-tabs=packages)
  - As far as I know it's exactly that library which includes the default `/etc/krb5.conf` file which also has the statement `includedir /etc/krb5.conf.d/` and some reasonable `[libdefaults]` settings

The full `krb5-libs` package is actually not required to have working Kerberos support. It suffices to add back the `/etc/krb5.conf`.
The `/etc/krb5.conf` will be picked up by Java when trying to login with Kerberos authentication.

### My solution
My personal solution was to copy the original `/etc/krb5.conf` and extend my `Dockerfile` to include this when we build our derived Keycloak 25 image.

**Comment by oers** [comment]
I'm affected by this as well. Maybe it would be a great idea to provide a docker image where the freeipa keycloak client is installed.

**Comment by ASzc** [comment]
Adding the whole krb5-libs package isn't feasible because of its dependency on openssl. My recommendation in previous cases was to install krb5-libs when customizing the image. I had assumed that JNA was used to access krb5-libs binaries, but apparently JGSS provides the needed functionality, only depending on its customary config files.

Adding in the default config files is feasible since they have no dependencies themselves, but I am surprised that they would be useful? You didn't configure any realm information?

If realm configuration is required, I would prefer to solve this via documentation. If customization is required anyway, then there's no point hacking krb5.conf into the keycloak image by itself

**Comment by hwanders** [comment]
@ASzc The default `krb5.conf` contains `includedir /etc/krb5.conf.d/` - so people can place custom config files in that folder without touching the `krb5.conf` file. That's why some setups don't require a customized `krb5.conf`, maybe they don't even knew about that file and were surprised when it was missing.

**Comment by ASzc** [comment]
Ok. I think this is best solved with a docs update then. Migration guide should mention it, and perhaps there's a spot in the main documentation this should be discussed too.

**Comment by ASzc** [comment]
Please see PR #45764
